SignalServiceKit/src/Messages/OWSMessageSender.m

```diff
@@ -118,47 +68,36 @@ - (void)setIsFatal:(BOOL)value
  * Used by `OWSMessageSender` to serialize message sending, ensuring that messages are emitted in the order they
  * were sent.
  */
-@interface OWSSendMessageOperation : NSOperation
+@interface OWSSendMessageOperation : OWSOperation
```

Comment:
Extracted the retry logic into a base class, which subclasses take advantage of by overriding hooks.

The base class is also used for the new Upload operation.

I feel like this is easier to read and reuse.